packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -268,7 +268,7 @@ export class EpochTrackerWithRedemption extends EpochTracker {
             if (fetchType === "treesLatest") {
```

Comment:
When there are no further calls to epochTracker.fetchAndParseAsJson, the joinSession call never returns as there is nothing to resolve/reject the treesLatestDeferral promise, so it just waits there forever. Should this reject the promise itself after some time?

Reply:
What is the actual sequence of events when it happens?
I can see three cases that I've listed in comments how we can get to joinSession:
1. create file -> join session
2. TreesLatest -> join session
3. getVersions -> load particular version -> join session

# 1 & # 3 if failed should never result in join session call.
# 2 is the only exception, as we could have parallel calls to TreesLatest & joinSession, and that's what this code tries to solve.

Reply:
I see, so 1 & 3 have to happen sequentially. And since TreesLatest is the only call that can happen in parallel, and it gets handled here, it's fine. But is it expected that no future parallel calls might get added?

Reply:
I guess the reason I'm not comfortable to resolve any failure here is that we do not know usage patters. Maybe someone will call getVersions and it fails, but then it will call TreesLatest & joinSession at the same time - we do not want to fail it just because something unrelated failed before.

Hard to predict the future. I think if we can ensure that even though TreesLatest & joinSession go in parallel, but TreesLatest always gets triggered first, then we would be able to make code safer by doing retries (and thus removing hangs for other scenarios) only if TreesLatest was already triggered. But I'm not sure if we can assume that - today we do not control that and they could be triggered in any order